### PR TITLE
QA-384: control_map_cleanup test fixes

### PR DESCRIPTION
**Pull request by oleorhagen**

### CODE REVIEW

**Comment by oleorhagen** [comment]
@mender-test-bot start pipeline

File: tests/meta-mender-ci/recipes-testing/mender-mock-server/files/mender-mock-server.py [hunk 1]
```diff
@@ -28,9 +29,18 @@
...
 
 class SimpleHTTPRequestHandler(BaseHTTPRequestHandler):
+
+    log = logging.getLogger(__name__)
+
     def do_GET(self):
         self.read_body()
 
+        if re.match(
+            "/api/devices/v2/deployments/device/deployments/([a-f0-9-]*)/update_control_map",
```
Comment (kacf) [inline tests/meta-mender-ci/recipes-testing/mender-mock-server/files/mender-mock-server.py:39]: Uhm, what about [this](https://github.com/mendersoftware/meta-mender/blob/ca2dd9b3062d2406a1159142e2f9e8f75f9bfb1f/tests/meta-mender-ci/recipes-testing/mender-mock-server/files/mender-mock-server.py#L55-L72)? Were we wrong, and it's already there?

**Comment by oleorhagen** [comment]
@mender-test-bot start pipeline

File: tests/meta-mender-ci/recipes-testing/mender-mock-server/files/mender-mock-server.py [hunk 2]
```diff
@@ -149,6 +155,28 @@ def do_GET_or_POST_deployments_next(self):
...
+            with open("/data/mender-mock-server-deployment-header.json", "rb") as body:
+                update_control_map = {
+                    "update_control_map": json.load(body).get("update_control_map")
+                }
+                if not update_control_map:
+                    self.log.critical(
+                        "do_GET_update_control_map: no update control map found in the '/data/mender-mock-server-deployment-header.json' file"
+                    )
+                    self.send_response(404)
+                um = json.dumps(update_control_map).encode(encoding="utf-8")
```
Comment (kacf) [inline tests/meta-mender-ci/recipes-testing/mender-mock-server/files/mender-mock-server.py:152]: Nitpick: `utf-8` is the default.

File: tests/meta-mender-ci/recipes-testing/mender-mock-server/files/mender-mock-server.py [hunk 3]
```diff
@@ -51,25 +51,6 @@ def do_GET(self):
...
-                    resp = json.dumps({
-                        "update_control_map": update["update_control_map"],
-                    }).encode("utf-8")
-                    self.send_response(200)
-                    self.send_header("Content-Type", "application/json")
-                    self.send_header("Content-Length", len(resp))
-                    self.end_headers()
-                    self.wfile.write(resp)
-            else:
-                self.send_empty_response(404)
```
Comment (kacf) [inline tests/meta-mender-ci/recipes-testing/mender-mock-server/files/mender-mock-server.py:62]: Nitpick: Should perhaps be squashed with the other one?

**Review by kacf — APPROVED** [review]
LGTM, just nitpicks.

**Comment by oleorhagen** [comment]
@mender-test-bot start pipeline

**Pull request merged**